### PR TITLE
fix: Achievements

### PR DESCRIPTION
**Pull request by kzalys**
Fixes the achievements. The problem was that the achievements system was using a `User` entity that is separate from the users on `hs_auth` and in my previous refactors I made that `User` entity redundant. I updated the achievements system to the users from `hs_auth`.

I did also end up breaking the push notification system but the achievements have higher priority right now.

### CODE REVIEW

File: src/controllers/achievementsController.ts [hunk 1]
```diff
@@ -64,24 +64,29 @@ export class AchievementsController implements AchievementsControllerInterface {
...
 
 			let achievements: Achievement[] = await this._achievementsService.getAchievements();
 			achievements = achievements.sort((a: Achievement, b: Achievement) => a.getTitle().localeCompare(b.getTitle()));
 
 			let prizesToClaim: AchievementProgress[] = await this._achievementsProgressService.getAchievementsProgressThatCanClaimPrize();
 			prizesToClaim = prizesToClaim.sort((a: AchievementProgress, b: AchievementProgress) => {
 				const achievementsComparison: number = a.getAchievement().getTitle().localeCompare(b.getAchievement().getTitle());
-				if (achievementsComparison === 0) {
-					return a.getUser().getName().localeCompare(b.getUser().getName());
+				if (achievementsComparison === 0 && usersMap.get(a.userId) !== undefined && usersMap.get(b.userId) !== undefined) {
```
Comment (amishshah) [inline src/controllers/achievementsController.ts:80]:
```suggestion
				if (achievementsComparison === 0 && typeof usersMap.get(a.userId) !== 'undefined' && typeof usersMap.get(b.userId) !== 'undefined') {
```

I think this is the safer way to do undefined checking in JS

**Review by seanjparker — APPROVED** [review]
LGTM

File: src/services/achievements/achievementsProgressService.ts [hunk 2]
```diff
@@ -32,20 +32,20 @@ export class AchievementsProgressService {
...
+   * @param userId The id of the user
    */
-	public getAchievementProgressForUser = async (achievement: Achievement, user: User): Promise<AchievementProgress> => {
+	public getAchievementProgressForUser = async (achievement: Achievement, userId: string): Promise<AchievementProgress> => {
 		let achievementProgress = await this._achievementsProgressRepository
 			.findOne({
 				where: {
 					achievementId: achievement.getId(),
-					user: user
+					userId: userId
```
Comment (amishshah) [inline src/services/achievements/achievementsProgressService.ts:42]:
```suggestion
					userId
```

Sorry :sweat_smile:

**Review by amishshah — CHANGES_REQUESTED** [review]

File: src/controllers/achievementsController.ts (hunk 1 above)
Comment (seanjparker) [reply src/controllers/achievementsController.ts:80]: I think that it true in JavaScript but in typescript it's okay to do !== undefined and the TSC will convert it to the safe JS check

**Review by amishshah — APPROVED** [review]

**Pull request merged**